lib/ansible/modules/cloud/amazon/elasticsearch.py

```diff
@@ -0,0 +1,421 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Ansible Project
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: elasticsearch
+short_description: Manage hosted Elasticsearch in Amazon Web Services.
+description:
+  - Manage hosted Elasticsearch in Amazon Web Services.
+  - Returns information about the hosted Elasticsearch.
+version_added: "2.6"
+requirements: [ boto3 ]
+author: "Zach Steindler (@steiza)"
+options:
+  domain:
+    description:
+      - Domain name for cluster.
+    required: true
+  state:
+    description:
+      - Create or destroy cluster
+    choices: [ "present", "absent" ]
+    required: true
+  version:
+    description:
+      - Version of Elasticsearch to use.
+    required: false
+    default: 5.1
+  instance_count:
+    description:
+      - Number of non-master nodes in the cluster.
+    required: false
+    default: 1
+  instance_type:
+    description:
+      - Instance type of non-master nodes in the cluster.
+    required: true
+  zone_awareness:
+    description:
+      - Should cluster be aware of AWS availability zones.
+    type: bool
+    required: false
+    default: true
+  dedicated_master_enabled:
+    description:
+      - If cluster have dedicated master nodes.
+    type: bool
+    required: false
+    default: false
+  dedicated_master_type:
+    description:
+      - Instance type of dedicated master nodes. Ignored if dedicated_master_enabled is False.
+    required: false
+  dedicated_master_count:
+    description:
+      - Number of dedicated master nodes. Ignored if dedicated_master_enabled is False.
```

Comment:
This statement is not true but it'd be good if it was.

If i pass `dedicated_master_enabled: false`  and still pass `dedicated_master_count` even if it's set to zero then i get an error 'The error was: botocore.errorfactory.ValidationException: An error occurred (ValidationException) when calling the UpdateElasticsearchDomainConfig operation: Dedicated master node count should be greater than 1.' 

This must mean that the parameter is still be passed to even though master_enabled is false.